### PR TITLE
add variational dropout to gluon ptb example

### PR DESCRIPTION
**Pull request by szha**
changes:
1. remove the download shell script
2. add cell api usage to the model
3. add variational dropout when using cell (requires #7067)

### CODE REVIEW

**Comment by szha** [comment]
@yzhang87 yes, reset is indeed called automatically. Removed. Thanks!

File: example/gluon/word_language_model/train.py [hunk 1]
```diff
@@ -70,8 +87,19 @@ def batchify(data, batch_size):
 
 
 ntokens = len(corpus.dictionary)
-model = model.RNNModel(args.model, ntokens, args.emsize, args.nhid,
-                       args.nlayers, args.dropout, args.tied)
+assert (not args.vardrop_out and not args.vardrop_state) or args.use_cell, \
+       'variational dropout is only available in cell API.'
+if not args.use_cell:
```
Comment (piiswrong) [inline example/gluon/word_language_model/train.py]: This looks too complicated for an example. not sure if we should add more comment or split it into to examples

Comment (szha) [inline example/gluon/word_language_model/train.py]: Alternative is duplicates. This is the simplest way to include example for RNN cells.

**Comment by piiswrong** [comment]
The word language model is intended to be a simple example, not a SOTA solution. I'm not sure we should keep adding more stuff to it.

**Comment by szha** [comment]
This is intended also as an example for using cell API, which we currently lack.

**Comment by piiswrong** [comment]
I think its better to use another example to introduce the cell API.
For example we can add an attention example

**Comment by szha** [comment]
OK

**Pull request closed without merging**